File: Chap3/Python/exo3.py
import os
#Exercice 2.3 (E) crire un programme qui crée 5 ﬁls en utilisant une boucle for. On remarquera que pour que le ﬁls ne crée pas lui-même plusieurs ﬁls, il faut interompre la boucle parun break dans le ﬁls.*/

for i in range(5):
    pid=os.fork()
    if pid==0:
        print("Processus",i)
        break
    else:
         os.wait()

File: Chap3/C/exo3.c
#include<stdio.h>
#include<unistd.h>
#include<sys/wait.h>

/*Exercice 2.3 (E) crire un programme qui crée 5 ﬁls en utilisant une boucle for. On remar-
quera que pour que le ﬁls ne crée pas lui-même plusieurs ﬁls, il faut interompre la boucle par
un break dans le ﬁls.*/

int main()
{
    pid_t pid_fils;

    for(int i=0;i<5;i++)
    {
       pid_fils=fork();
       if(pid_fils==0)
       {
        printf("Le processus fils numéro %d\n",i);
        break;
       }
       else
       {
         waitpid(pid_fils,NULL,0);
       }    
    }
    return 0;
}
